src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -1732,6 +1753,24 @@ private IndexAnalysisSettings2 readAnalysisSettings() throws IOException {
         return dao.read(reader);
     }
 
+    private boolean checkXrefExistence(String path) {
+        RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+        boolean chkres = whatXrefFile(path, env.isCompressXref()).exists();
+
+        if (!env.isGenerateHtml()) {
+            if (chkres) {
+                LOGGER.log(Level.FINEST, "Extraneous {0}", path);
+                removeXrefFile(path);
```

Comment:
I'd just leave the file alone. Firstly, it does not go well with the name of the method ("check existence" should do just what it says in the name), secondly it can be a result of current mis-configuration.